### ISSUE TITLE
Handling for URLs containing IPv6 addresses broken

### ISSUE THREAD

**Issue by gjasny** [yhirose/cpp-httplib#979]
Hello,

I noticed that the library cannot handle URLs like `http://[::1]/` because it does not consider the `[]`-escaped IPv6 addresses. It's a problem in both: the Client constructors as well as the `Host:`-HTTP header.

Thanks,
Gregor

**Comment by yhirose** [comment]
@gjasny, thanks for the report. I right now don't have time to work on it. If you have time, could you send me a pull request with a unit test case? Thanks!

**Comment by gjasny** [comment]
Hello,

thanks for taking care. Unfortunately the `Host` header still looks wrong:

```
GET /cache/8e193hu1ntu7bvd6ohhsrncr94hibhn8s HTTP/1.1
Accept: */*
Host: ::1:8080
User-Agent: ccache/master.b7e6280e
```

Maybe you could use a dedicated URL class for parsing like: [CxxUrl](https://github.com/chmike/CxxUrl)?

Thanks,
Gregor

**Comment by lifw555** [comment]
ipv6 must have [ ] in url or header

**Comment by yhirose** [comment]
Open this issue again. A pull request is welcome!

**Comment by Mufan187569** [comment]
Recently, I have encountered some minor issues while self-studying network transmission and using CPP-httplib. I would like to ask for your advice. The main problem is that when I use the cpp-httplib library for HTTP transmission, there are no issues. But I chose to compile on both the client and server: # define SERVER_ CERT_ When using FILE, # ifdef, # else, and # endif, both HTTP and HTTPS transmissions cannot be used. I don't know why?
